source/poodinis/container.d

```diff
@@ -192,6 +201,44 @@ synchronized class DependencyContainer {
         return newRegistration;
     }
 
+    /**
+     * 
+     */
+    Registration register(SuperType, ConcreteType : SuperType)(InjectionInitializer!ConcreteType creator, 
+            RegistrationOption options = RegistrationOption.none) if (!is(ConcreteType == struct)) {
+
+        TypeInfo registeredType = typeid(SuperType);
+        TypeInfo_Class concreteType = typeid(ConcreteType);
+
+        debug(poodinisVerbose) {
+            writeln(format("DEBUG: Register type %s (as %s)", concreteType.toString(), registeredType.toString()));
+        }
+
+        auto existingRegistration = getExistingRegistration(registeredType, concreteType);
+        if (existingRegistration) {
+            return existingRegistration;
+        }
+
+        InstanceFactory instanceFactory = new class InstanceFactory {
+            protected override Object createInstance() { 
+                return creator();
+            }
+        };
+
+        auto newRegistration = new AutowiredRegistration!ConcreteType(registeredType, instanceFactory, this);
+        newRegistration.singleInstance();
```

Comment:
This doesn't make much sense anymore, the custom instance factory doesn't guarentee a singleInstance anymore.
Instead of supplying the factory in the signature of register(), is is possible to supply the factory like singleInstance(), newInstance() etc does?

Example:
`dependencies.register!Driver.createdBy( {
	return new Driver();
})`

Not particularly married to the "createdBy" name